### PR TITLE
Added `list-no-top-level-material-ui-4-imports` script

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Added a `list-no-top-level-material-ui-4-imports` script that will list all the frontend plugins that aren't using the `no-top-level-material-ui-4-imports` ESLint rule so that we can slowly start adding it and track the progress.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: scripts/verify-no-top-level-material-ui-4-imports.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const { getPackages } = require('@manypkg/get-packages');
+const { resolve, join } = require('path');
+
+async function main() {
+  const rootPath = resolve(__dirname, '..');
+  const { packages } = await getPackages(rootPath);
+
+  for (const pkg of packages) {
+    pkgRole = pkg.packageJson.backstage?.role;
+    if (pkgRole === 'frontend-plugin') {
```
Comment (freben) [inline scripts/verify-no-top-level-material-ui-4-imports.js:28]: should this also include `web-library`?

File: scripts/verify-no-top-level-material-ui-4-imports.js
Comment (freben) [inline scripts/verify-no-top-level-material-ui-4-imports.js:1]: arguably the name of this script implies that it does verification - maybe a light tweak to the name

File: scripts/verify-no-top-level-material-ui-4-imports.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      const eslintrcPath = join(pkg.dir, '.eslintrc.js');
+      const targetEslintrc = (await fs.readFile(eslintrcPath)).toString();
+      if (
+        targetEslintrc.includes(
+          "'@backstage/no-top-level-material-ui-4-imports': 'error'",
+        )
+      ) {
+        continue;
```
Comment (freben) [inline scripts/verify-no-top-level-material-ui-4-imports.js:36]: maybe better to invert the condition and log inside here, skipping the continue

Comment (awanlin) [reply scripts/verify-no-top-level-material-ui-4-imports.js]: Sure, done 👍

File: scripts/verify-no-top-level-material-ui-4-imports.js (hunk 1 above)
Comment (awanlin) [reply scripts/verify-no-top-level-material-ui-4-imports.js]: Makes sense, I've added this 👍

File: scripts/verify-no-top-level-material-ui-4-imports.js
Comment (Rugvip) [reply scripts/verify-no-top-level-material-ui-4-imports.js:1]: Thinking `list-no-top-level-material-ui-4-imports.js`?

**Review by Rugvip — APPROVED** [review]
Nice! 👍 

Looks good at such, but do we need it merged into the repo still? How far along is the migration? Then again this might be useful to keep around for the future and community plugins repo anyway

File: scripts/verify-no-top-level-material-ui-4-imports.js
Comment (awanlin) [reply scripts/verify-no-top-level-material-ui-4-imports.js:1]: Sounds good, done!

**Comment by awanlin** [comment]
@Rugvip @freben I've updated the name, there are still some things that have not been migrated yet so still think this makes sense to add. Can I get another approval and merge, please?

File: scripts/list-no-top-level-material-ui-4-imports.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+async function main() {
+  const rootPath = resolve(__dirname, '..');
+  const { packages } = await getPackages(rootPath);
+
+  for (const pkg of packages) {
+    pkgRole = pkg.packageJson.backstage?.role;
+    if (pkgRole === 'frontend-plugin' || pkgRole === 'web-library') {
+      const depKeys = Object.keys(pkg.packageJson.dependencies);
+      if (depKeys.findIndex(d => d.includes('@material-ui')) !== -1) {
```
Comment (freben) [inline scripts/list-no-top-level-material-ui-4-imports.js:30]:
```suggestion
      if (depKeys.some(d => d.includes('@material-ui'))) {
```

bit simpler, but just a hint for the future

**Review by freben — APPROVED** [review]

**Pull request merged**